plugins/BEdita/Core/src/Model/Action/SetRelatedObjectsAction.php

```diff
@@ -95,10 +39,14 @@ protected function update(EntityInterface $entity, $relatedEntities)
             return $action->execute(compact('entity', 'relatedEntities'));
         }
 
-        $relatedEntities = $this->prepareRelatedEntities($relatedEntities, $entity);
+        $relatedEntities = $this->diff($entity, $relatedEntities, true, $affectedEntities);
 
-        $diff = $this->diff($entity, $relatedEntities);
+        if (!$this->Association->replaceLinks($entity, $relatedEntities)) {
+            return false;
+        }
 
-        return $this->Association->replaceLinks($entity, $relatedEntities) ? $diff : false;
+        return collection($affectedEntities)
```

Comment:
parent abstract method in `UpdateAssociatedAction` has return type `@return int|false` here we have `array`, which one is correct?

Reply:
Both are correct 😄 that also means none of them actually is correct. 😱

The return type for `UpdateAssociatedAction` should be `int|array|false`.